File: Models/Event.swift
import SwiftUI

struct Event: Identifiable, Hashable {
    let id: UUID
    let title: String
    let start: Date
    let end: Date
    let location: String?
    let attendees: [User]
    let color: Color
    
    init(title: String, start: Date, end: Date, location: String? = nil, attendees: [User] = [], color: Color) {
        self.id = UUID()
        self.title = title
        self.start = start
        self.end = end
        self.location = location
        self.attendees = attendees
        self.color = color
    }
}

File: src/models/Event.ts
import { User } from './User';

export interface Event {
  id: string;
  title: string;
  start: Date;
  end: Date;
  location?: string;
  attendees: User[];
  color: string;
}

export const createEvent = (
  title: string,
  start: Date,
  end: Date,
  location?: string,
  attendees: User[] = [],
  color: string = '#E5E7EB'
): Event => ({
  id: Math.random().toString(36).substring(2, 15) + Math.random().toString(36).substring(2, 15),
  title,
  start,
  end,
  location,
  attendees,
  color,
});
